lib/section.zsh

```diff
@@ -46,9 +46,18 @@ _compose_prompt() {
   # http://zsh.sourceforge.net/Doc/Release/Conditional-Expressions.html
   setopt EXTENDED_GLOB LOCAL_OPTIONS
 
-  # Threat the first argument as list of prompt sections
+  # Treat the first argument as list of prompt sections
   # Compose whole prompt from diferent parts
+  # If section is a function then invoke it
+  # If the composed name spaceship_$section is a function invoke it
   for section in $@; do
-    spaceship_$section
+    if typeset -f + $section &> /dev/null; then
+      eval $section
+      continue
+    fi
+    if typeset -f + "spaceship_$section" &> /dev/null; then
```

Comment:
I think we should get rid of the branch, and just require that plugin functions be prefixed with `spaceship_`.

Reply:
I would not favor that approach, because many functions are already declared by other libraries script, and do not follow that convention or course. Typically with oh-my-zsh, lots of plugins exists, you just have to invoke the right function inside the prompt var of the theme.

Requiring functions to be prefixed by spaceship is not quite flexible.

I'm not against the concept of plugins though, this may be a complementary approach.

Reply:
> Requiring functions to be prefixed by spaceship is not quite flexible.

It just requires adding a wrapper function, which I don't think is a big burden. I see benefits since it keeps spaceship core simple (custom functions are loaded with the same control flow).

Reply:
I think this approach is not elegant. If the software can do it for free in a generic manner why imposing this to users.